### DISCUSSION TITLE
Got-scraping doesn't throw an error for error 4xx.

### DISCUSSION THREAD

**Discussion by meotimdihia** [apify/got-scraping#82] [Q&A] [answered]
Is this intent? it is different with `got`.

**Comment by B4nan** [comment, marked as answer]
Yes, [intentional](https://github.com/apify/got-scraping/blob/15744b41a11a97069d232deeccd25d97e597dc6b/src/index.ts#L32), I don't recall the exact reason now, but most probably to have more similar behavior to browsers. The library is mainly developed for making requests in http crawlers in [crawlee](https://github.com/apify/crawlee).
